src/org/openstreetmap/josm/actions/search/SearchAction.java

```diff
@@ -348,10 +376,44 @@ public boolean isValid() {
             }
         });
 
-        final JPanel p = new JPanel(new GridBagLayout());
+        /*
+         * Setup the logic to append preset queries to the search text field according to
+         * selected preset by the user. Every query is of the form ' group/sub-group/.../presetName'
+         * if the corresponding group of the preset exists, otherwise it is simply ' presetName'.
+         */
+        TaggingPresetSelector selector = new TaggingPresetSelector(false, false);
+        selector.setBorder(BorderFactory.createTitledBorder(tr("Search by preset")));
+
+        selector.setDblClickListener(ev -> {
+            TaggingPreset selectedPreset = selector.getSelectedPresetAndUpdateClassification();
+
+            if (selectedPreset == null) {
+                return;
+            }
+
+            /*
+             * Make sure that the focus is transferred to the search text field, in order
+             * to allow the user simply delete autocompleted part of the query.
+             */
+            editorComponent.requestFocusInWindow();
+
+            SwingUtilities.invokeLater(() -> {
```

Comment:
I know that, but the idea was to postpone the insertion event after focus is acquired.